packages/shared/src/components/notifications/NotificationItemAvatar.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { ReactElement } from 'react';
+import SourceButton from '../cards/SourceButton';
+import { ProfilePicture } from '../ProfilePicture';
+
+export enum NotificationAvatarType {
+  User = 'user',
+  Source = 'source',
+}
+
+export interface NotificationItemAvatarProps {
+  type: NotificationAvatarType;
+  image: string;
+  name: string;
+  targetUrl: string;
+  referenceId: string;
+}
+
+function NotificationItemAvatar({
+  type,
+  image,
+  name,
+  referenceId,
+}: NotificationItemAvatarProps): ReactElement {
+  if (type === NotificationAvatarType.Source) {
+    return <SourceButton source={{ id: referenceId, name, image }} />;
+  }
+
+  return <ProfilePicture user={{ image, id: referenceId }} />;
```

Comment:
Generic question:

Should we maybe hard cast a type === here as well?
In case we introduce a new one as we go, rather have that render as empty?

Reply:
Very good point. We shall do that.